openapi.yaml

```diff
@@ -23,901 +31,1078 @@ tags:
-          schema:
-            $ref: '#/definitions/imsx_StatusInfoDType'
+          $ref: '#/responses/400'
         '401':
-          description: 'The request was not correctly authorised i.e. ''unauthorisedrequest''. This would be accompanied by the ''codeMajor/severity'' values of ''failure/error''. '
-          schema:
-            $ref: '#/definitions/imsx_StatusInfoDType'
+          $ref: '#/responses/401'
         '403':
-          description: 'This is used to indicate that the server can be reached and process the request but refuses to take any further action i.e. ''forbidden''. This would be accompanied by the ''codeMajor/severity'' values of ''failure/error''. '
-          schema:
-            $ref: '#/definitions/imsx_StatusInfoDType'
+          $ref: '#/responses/403'
+        '404':
+          $ref: '#/responses/404'
         '422':
-          description: 'This error condition may occur if a JSON request body contains well-formed (i.e. syntactically correct), but semantically erroneous, JSON instructions. This would be accompanied by the ''codeMajor/severity'' values of ''failure/error''. '
-          schema:
-            $ref: '#/definitions/imsx_StatusInfoDType'
+          $ref: '#/responses/422'
         '429':
-          description: 'The server is receiving too many requests i.e. ''server_busy''. Retry at a later time. This would be accompanied by the ''codeMajor/severity'' values of ''failure/error''. '
-          schema:
-            $ref: '#/definitions/imsx_StatusInfoDType'
+          $ref: '#/responses/429'
         '500':
-          description: 'This code should be used only if there is catastrophic error and there is not a more appropriate code i.e. ''internal_server_error''. This would be accompanied by the ''codeMajor/severity'' values of ''failure/error''. '
-          schema:
-            $ref: '#/definitions/imsx_StatusInfoDType'
+          $ref: '#/responses/500'
         default:
-          description: 'This is the default error response when no other appropriate code is available.  This would be accompanied by the ''codeMajor/severity'' values of ''failure/error''. '
-          schema:
-            $ref: '#/definitions/imsx_StatusInfoDType'
+          $ref: '#/responses/default'
+parameters:
+  limit:
+    name: page[limit]
+    in: query
+    description: To define the download segmentation value i.e. the maximum number of records to be
+      contained in the response.
+    required: false
+    type: integer
+    format: int32
+    default: 100
+    minimum: 1
+    allowEmptyValue: false
+  offset:
+    name: page[offset]
+    in: query
+    description: The number of the first record to be supplied in the segmented response message.
+    required: false
+    type: integer
+    format: int32
+    default: 0
+    minimum: 0
+    allowEmptyValue: false
+  sort:
```

Comment:
do we want to follow the upper camel case rule for parameter names?

Reply:
Don't we usually do lower camel case for parameters and upper camel case for definitions? I don't think we have any conventions in our [style guide](https://github.com/osu-mist/style-guides/blob/master/OpenAPI.md) so I can add something there after we decide.

It looks like the skeleton has lower camel case [here](https://github.com/osu-mist/express-api-skeleton/blob/f4fceb43c28d23811c943c1c3d731a712a6a435d/openapi.yaml#L80-L103) and the openapi 2 example does something similar [here](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameters-definition-object-example). What do you think is the best solution? In my mind, I've always thought of it as parameters being like a variable and definitions being like a class/object, what are your thoughts on this?

Reply:
Oh, after looking at your changes, I think I misunderstood. In my previous comment, I was talking about the names we give to our `definitions`. As far as the actual query parameter names, `filter`, `sort`, and `fields` are reserved by jsonapi so we should use those names. If we want to add implementation-specific query parameters that don't belong to any reserved query parameter family, that's when we need the requirement of at least one non a-z character.